python/test/conftest.py

```diff
@@ -136,3 +138,56 @@ def tempdir(request):
 
     """
     return _create_tempdir(request)
+
+
+@pytest.fixture(scope="function")
+def cg_solver():
+    """Simple Conjugate Gradient solver for SPD problems,
+       which can work in serial or parallel for testing use.
+       Not suitable for large problems."""
+
+    # Basic Conjugate Gradient solver
+    def _cg(comm, A, b, x):
+        kmax = 500
+        rtol2 = 10 * np.finfo(x.array.dtype).eps
```

Comment:
This is very tight. 

Should probably be a parameter with a default value.

Reply:
Note this is rtol^2, so about 1e-8 for double. I will make it into a parameter